tests/unit/parsers/test.manifestjson.js

```diff
@@ -3185,4 +3185,337 @@ describe('ManifestJSONParser', () => {
+        },
+      ],
+      [
+        'invalid strict_min_version (version below the strict min)',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '88.0',
+              },
+            },
+            permissions: ['alarms'],
+          },
+          restrictedPermissions: new Map([['alarms', '88.1.3']]),
+        },
+      ],
+      [
+        'invalid strict_min_version (version below the strict min)',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '88.1.2',
+              },
+            },
+            permissions: ['alarms'],
+          },
+          restrictedPermissions: new Map([['alarms', '88.1.3']]),
+        },
+      ],
+      [
+        'invalid strict_min_version (version below the strict min) with unrelated restrictions',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '88.1.2',
+              },
+            },
+            permissions: ['alarms'],
+          },
+          restrictedPermissions: new Map([
+            ['alarms', '88.1.3'],
+            [('bookmarks', '100.0')],
+          ]),
+        },
+      ],
+      [
+        'multiple permissions and version below the strict min',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '88.0',
+              },
+            },
+            permissions: ['bookmarks', 'alarms'],
+          },
+          restrictedPermissions: new Map([['alarms', '88.1.3']]),
+        },
+      ],
+      [
+        'uppercase permission and version below the strict min',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '88.0',
+              },
+            },
+            permissions: ['ALARMS'],
+          },
+          restrictedPermissions: new Map([['alarms', '88.1.3']]),
+        },
+      ],
+      [
+        'versions contain non-numeric chars',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '1.1pre1a',
+              },
+            },
+            permissions: ['ALARMS'],
+          },
+          restrictedPermissions: new Map([['alarms', '1.1pre1aa']]),
+        },
+      ],
+    ])(
+      'reports an error when: %s',
+      (title, { manifestProps, restrictedPermissions }) => {
+        const { parser, linter } = validate({
+          manifestProps,
+          restrictedPermissions,
+        });
+        const { errors } = linter.collector;
+
+        expect(parser.isValid).toEqual(false);
+        expect(errors.length).toEqual(1);
+        expect(errors[0].code).toEqual(messages.RESTRICTED_PERMISSION);
```

Comment:
I don't think we care that the error collect is only one, as long as we got the expected restricted permission one.
I also prefer if we could use `expect(...).toEqual(expect.arrayContaining([...]))` to assert the errors that we expect are there, because when that isn't the case it produces a better error message (which include both the expected values and the actual array elements that didn't match the expectation), while this one wouldn't provide any useful information to investigate a CI job that is failing this test:

```diff
diff --git a/tests/unit/parsers/test.manifestjson.js b/tests/unit/parsers/test.manifestjson.js
index bda9f550..c9e741c9 100644
--- a/tests/unit/parsers/test.manifestjson.js
+++ b/tests/unit/parsers/test.manifestjson.js
@@ -3327,8 +3327,14 @@ describe('ManifestJSONParser', () => {
         const { errors } = linter.collector;

         expect(parser.isValid).toEqual(false);
-        expect(errors.length).toEqual(1);
-        expect(errors[0].code).toEqual(messages.RESTRICTED_PERMISSION);
+        expect(errors).toEqual(
+          expect.arrayContaining([
+            expect.objectContaining({
+              code: messages.RESTRICTED_PERMISSION,
+              file: 'manifest.json',
+            }),
+          ])
+        );
       }
     );
```